docs/dev/frontend.rst

```diff
@@ -87,44 +75,88 @@ As can be seen above, Kolibri modules are defined in the root directory of the '
 when it is ready.
 
+
+The Kolibri Core Javascript App
+-------------------------------
+
+The Kolibri Core Javascript App exposes commonly used libraries and acts as the mediator and point of communication
+between different Kolibri Modules.
+
+These methods are publicly exposed methods of the Mediator class::
+
+  kolibriGlobal.register_kolibri_module_async - Register a Kolibri module for asynchronous loading.
+  kolibriGlobal.register_kolibri_module_sync - Register a Kolibri module once it has loaded.
+  kolibriGlobal.stop_listening - Unbind an event/callback pair from triggering.
+  kolibriGlobal.emit - Emit an event, with optional args.
+
+In addition, the lib property exposes the following libraries::
+
+  kolibriGlobal.lib.vue - the global Vue module for Kolibri
+  kolibriGlobal.lib.loglevel - the global logging module for Kolibri
+  kolibriGlobal.lib.coreBase - the glboal component library for common UI components for Kolibri
+
+Finally, once loaded into the frontend, we embellish the kolibriGlobal object with the following object::
+
+  kolibriGlobal.urls
+
+This object is defined by `Django JS Reverse <https://github.com/ierror/django-js-reverse>`_ and exposes Django URLs
+on the client side. This will primarily be used for accessing API Urls for synchronizing with the REST API. See the
+Django JS Reverse documentation for details on invoking the Url.
+
+
 Writing Kolibri Modules
 -----------------------
 
 All Kolibri Modules should extend the KolibriModule class found in
-`kolibri/plugins/assets/src/kolibri_module/kolibri_module.js`. For convenience this can be referenced in a module with
+`kolibri/core/assets/src/kolibri_module.js`. For convenience this can be referenced in a module with
 the following syntax::
 
-    var KolibriModule = require('kolibri_module');
+    const KolibriModule = require('kolibri_module');
+    const logging = require('loglevel');
+
+    class ExampleModule extends KolibriModule {
 
-    var ExampleModule = KolibriModule.extend({
+        initialize: () => {
+            logging.info('Doing something before the Module is registered with the Core App!');
+        }
+
+        ready: () => {
+            logging.info('Module is registered and ready to do things!');
+        }
 
-        hello_world: function(message) {
+        hello_world: (message) => {
             logging.info('Hello world!', message);
-        },
+        }
 
-        goodbye_world: function(message) {
+        goodbye_world: (message) => {
             logging.info('Goodbye, cruel world!', message);
         }
-    });
+    }
 
 The methods defined above are the ones that can be referenced in the `events` and `once` properties of the plugin that
 defines the Kolibri Module. Defining it in this way allows for asynchronous loading and registration without having to
 load or execute the Javascript code.
 
+The ready method will be automatically executed once the Module is loaded and registered with the Kolibri Core App.
```

Comment:
Okay, I still think that we should establish the convention that the DOM is not fully loaded when this method is executed, even if it's *mostly* loaded since the order in which we inject these scripts might change, deliberately or otherwise. Moreover it creates a simpler mental mode if we can just say something very specific like

> Assume `ready` is executed *after* the core module's `ready` and *before* the DOM is loaded 

vs

> the DOM might be *mostly* loaded... check where that bundle is inserted in the template to determine the order.